hashintel/hash
# Migrate from using the mini_v8 bindings to Deno's rusty_v8

## 🌟 What is the purpose of this PR?

Reduce maintainer burden as we won't have to maintain our own mini_v8.
Simplify the engine build process, no need to pull v8.
A more up-to-date v8 means more features, fast compile-time and less bugs.

## 🐾 Next steps

Get rusty_v8 from crates.io once https://github.com/denoland/rusty_v8/pull/926 is merged and a new release is made.

## 🔍 What does this change?

- Replace mini_v8 by rusty_v8
- Upgrade v8 from v8.4 to v10.0

## 📜 Does this require a change to the docs?

Yes, the build instructions in the README changed.

## 🔗 Related links

[Asana task (upgrade v8)](https://app.asana.com/0/1199548034582004/1201907929549868/f) _(internal)_
[Asana task (switch library)](https://app.asana.com/0/1201893074552404/1201481007343146/f) _(internal)_
[v8 update new API](https://docs.google.com/document/d/1sTc_jRL87Fu175Holm5SV0kajkseGl2r8ifGY76G35k/)
## 🛡 What tests cover this?

- All tests using the JS runner

## ❓ How to test this?

1.  Checkout the branch
2.  Try `cargo test`
3.  Confirm that no test fail
